src/main/kotlin/no/nav/syfo/api/model/NewVurderingRequestDTO.kt

```diff
@@ -1,13 +1,49 @@
 package no.nav.syfo.api.model
 
-import no.nav.syfo.domain.VurderingType
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeName
 import no.nav.syfo.domain.DocumentComponent
 import java.time.LocalDate
 
-data class NewVurderingRequestDTO(
-    val personident: String,
-    val vurderingType: VurderingType,
-    val begrunnelse: String,
-    val document: List<DocumentComponent>,
-    val varselSvarfrist: LocalDate?,
-)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "vurderingType")
```

Comment:
Forstår jeg det riktig som at vi her injecter feltet `vurderingType` som settes vha `@JsonTypeName("FORHANDSVARSEL")` etc for hver klasse? Hva er evt fordelen med dette vs å bare definere `vurderingType` for hver av de direkte som en `val`?

Reply:
Stemmer! Hvis vi definerer `vurderingType` direkte som `val` klarer ikke jackson serialisereren å finne ut hvilken type av `NewVurderingRequest` den skal serialisere. Slik jeg har forstått det setter vi et felt her som skal fungere som en diskriminator for hvilken av typene vi mottar.
I endepunktet har vi `call.receive<NewVurderingRequestDTO>()`, og da må jackson vite hvilken av subtypene til `NewVurderingRequestDTO` som den skal begynne å serialisere.

Reply:
Aha skjønner, da gir det mening 👍🏼